docs/faq.md

```diff
@@ -0,0 +1,25 @@
+# Frequently Asked Questions
+
+## Why do I have to run `make generate` in the beginning?
+
+Most software projects use add-ons and libraries to make development easier (one popular add-on for package management in the iOS development world is **[Cocoapods](http://cocoapods.org)**). Using package managers makes working with 3rd-party libraries easier. Projects like this normally require additional setup and or configuration.
+
+To save time, the setup and build process is usually automated. This project uses **[GNU Make](https://www.gnu.org/software/make/)**. At the time of writing the setup chain defines the following [rules](https://www.gnu.org/software/make/manual/html_node/Makefile-Contents.html#Makefile-Contents) that do the following things (defined in the [Makefile](https://github.com/kgellci/Area51/blob/master/Makefile)):
+
+1. Rule: `generate`
+    - Creates a `user.xconfig` file that allows different developers to enter their individual configurations (it's gitignored)
+    - Runs two scripts (`ensure-xcodegen.sh`, `ensure-swiftlint.sh`) that install Xcodegen and Swiftlint respectively.
+    - Runs Xcodegen
+
+2. Rule: `lint`
+    - Runs a script
+    - Runs SwiftLint
+
+The Area51 team uses **[XcodeGen](https://github.com/yonaskolb/XcodeGen)** to generate the `Area51.xcodeproj` file to make sure there are no merge conflicts when multiple developers are adding files. And **[SwiftLint](https://github.com/realm/SwiftLint)** to enforce a code style which is standard across many companies.
+
+## The project uses Xcodegen, but instead of `xcodegen generate` it uses `make generate`. Why?
```

Comment:
Would add that `make generate` will fetch the required version of xcodegen if needed.